src/mapping/GinkgoRadialBasisFctSolver.hpp

```diff
@@ -0,0 +1,533 @@
+// Ginkgo Data Structures
+using GinkgoVector = gko::matrix::Dense<>;
+using GinkgoMatrix = gko::matrix::Dense<>;
+using GinkgoScalar = gko::matrix::Dense<>;
+// Ginkgo Solver
+using cg    = gko::solver::Cg<>;
+using gmres = gko::solver::Gmres<>;
+using mg    = gko::solver::Multigrid;
+using ir    = gko::solver::Ir<>;
+// Ginkgo Preconditioner
+using jacobi   = gko::preconditioner::Jacobi<>;
+using cholesky = gko::preconditioner::Ic<>;
+using ilu      = gko::preconditioner::Ilu<>;
+
+// Ginkgo Helpers
+using amgx_pgm = gko::multigrid::AmgxPgm<>; // TODO: It was later renamed to Pgm so this needs to be fixed as soon as switching to newer Ginkgo version is done
+
+enum SolverType {
+  CG,
+  GMRES,
+  MG
+};
+
+enum PreconditionerType {
+  Jacobi,
+  Cholesky,
+  Ilu,
+  None
+};
+
+const std::map<std::string, SolverType> solverTypeLookup{
+    {"cg-solver", SolverType::CG},
+    {"gmres-solver", SolverType::GMRES},
+    {"mg-solver", SolverType::MG}};
+
+const std::map<std::string, PreconditionerType> preconditionerTypeLookup{
+    {"jacobi-preconditioner", PreconditionerType::Jacobi},
+    {"cholesky-preconditioner", PreconditionerType::Cholesky},
+    {"ilu-preconditioner", PreconditionerType::Ilu},
+    {"no-preconditioner", PreconditionerType::None}};
+
+const std::map<std::string, std::function<std::shared_ptr<gko::Executor>()>> ginkgoExecutorLookup{{"reference-executor", [] { return gko::ReferenceExecutor::create(); }},
+                                                                                                  {"omp-executor", [] { return gko::OmpExecutor::create(); }},
+                                                                                                  {"cuda-executor", [] { return gko::CudaExecutor::create(0, gko::OmpExecutor::create(), true, gko::allocation_mode::device); }},
+                                                                                                  {"hip-executor", [] { return gko::HipExecutor::create(0, gko::OmpExecutor::create(), true); }}};
+
+template <typename RADIAL_BASIS_FUNCTION_T>
+class GinkgoRadialBasisFctSolver {
+public:
+  GinkgoRadialBasisFctSolver() = default;
+
+  GinkgoRadialBasisFctSolver(const MappingConfiguration::GinkgoParameter &ginkgoParameter);
+
+  /// Assembles the system matrices and computes the decomposition of the interpolation matrix
+  template <typename IndexContainer>
+  GinkgoRadialBasisFctSolver(RADIAL_BASIS_FUNCTION_T basisFunction, const mesh::Mesh &inputMesh, const IndexContainer &inputIDs,
+                             const mesh::Mesh &outputMesh, const IndexContainer &outputIDs, std::vector<bool> deadAxis, Polynomial polynomial,
+                             const MappingConfiguration::GinkgoParameter &ginkgoParameter);
+
+  ~GinkgoRadialBasisFctSolver() = default;
+
+  GinkgoRadialBasisFctSolver(const GinkgoRadialBasisFctSolver &solver) = delete;
+  GinkgoRadialBasisFctSolver &operator=(const GinkgoRadialBasisFctSolver &solver) = delete;
```

Comment:
I added them because I also added constructor and deconstructor. In my eyes, copying this class is not necessary in preCICE. Do you think it is? If so, I can remove it of course.